src/core/ucc_coll.c

```diff
@@ -193,10 +193,12 @@ UCC_CORE_PROFILE_FUNC(ucc_status_t, ucc_collective_init,
         task->cb = coll_args->cb;
         task->flags |= UCC_COLL_TASK_FLAG_CB;
     }
+    task->seq_num = team->seq_num++;
```

Comment:
do you need atomic inc here?

Reply:
not needed i think since multiple threads are not allowed to call collective_init concurrently on the same team.

Reply:
why did we do in HCOLL? Ex: https://github.com/Mellanox/hcoll/blob/master/src/hcoll/ml/coll_ml_bcast.c#L1356

Reply:
it came from coll/ml. i think it was not necessary.

Reply:
I'm not sure, if we had well-defined thread model there, particular from ML POV.